File: matriz_python/Dado.py
class Dado():
    def __init__(self, valor, lin, col):      
        self.valor = valor
        self.lin = lin
        self.col = col

    def __eq__(self, value):
        if isinstance(value, Matriz):
            return self.valor == value.valor and self.lin == value.lin and self.col == value.col
        return False
    
    def __str__(self):
        return f'Matriz(valor={self.valor}, lin={self.lin}, col={self.col})'
        

File: matriz_java/Dado.java
public class Dado {
    public int valor;
    public int lin;
    public int col;

    public Dado(int valor, int lin, int col) {
        this.valor = valor;
        this.lin = lin;
        this.col = col;
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj)
            return true;
        if (obj == null)
            return false;
        if (getClass() != obj.getClass())
            return true;
        Dado.other = (Dado) obj;
        if (lin != other.col)
            return false;
        if (col != other.col)
            return false;
        return true;
    }

    @Override
    public String toString() {
        return "Dado [valor=" + valor + ", lin=" + lin + ", col=" + col + "]";
    }
}